### ISSUE TITLE
Error: 'LateInitializationError' isn't a type

### ISSUE THREAD

**Issue by tiholic** [flutter/flutter#75527]
## Steps to Reproduce

1. Run `flutter create bug`.
2. Update the files as follows: 
```dart

void main() {
  TryingToBeLate tryingToBeLate = TryingToBeLate();
  try {
    print(tryingToBeLate.lateString);
  } on LateInitializationError {
    print("Oops!");
  }
  runApp(MyApp());
}

// ...default code...


class TryingToBeLate {

  late final String lateString;

  TryingToBeLate(){
    // But never initializing it
  }

}
```
3. Run the app

**Expected results:**
App runs without errors

**Actual results:**
Dart error: `Error: 'LateInitializationError' isn't a type.`

<details>
  <summary>Logs</summary>

```logs
...
[ +481 ms] Launching lib/main.dart on Chrome in debug mode...
[ +215 ms] Updating assets
[ +174 ms] Waiting for connection from debug service on Chrome...
[ +332 ms] <- reset
[  +24 ms] /Users/rohit/Applications/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/Users/rohit/Applications/flutter/bin/cache/artifacts/engine/darwin-x64/frontend_server.dart.snapshot --sdk-root
/Users/rohit/Applications/flutter/bin/cache/flutter_web_sdk/ --incremental --target=dartdevc --debugger-module-names --experimental-emit-debug-metadata
-DFLUTTER_WEB_AUTO_DETECT=true --output-dill /var/folders/ks/8rd8hm0n18b3wh8dzchfc7400000gp/T/flutter_tools.xvq1f5/flutter_tool.Vt9kjc/app.dill
--libraries-spec file:///Users/rohit/Applications/flutter/bin/cache/flutter_web_sdk/libraries.json --packages
/Users/rohit/workspace/infitio/chartered_ui/bug/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts
--track-widget-creation --filesystem-root /var/folders/ks/8rd8hm0n18b3wh8dzchfc7400000gp/T/flutter_tools.xvq1f5/flutter_tools.9Alhly --filesystem-scheme
org-dartlang-app --initialize-from-dill build/b1b715402d823b7fd5c2b68d2edcb2ce.cache.dill.track.dill --platform
file:///Users/rohit/Applications/flutter/bin/cache/flutter_web_sdk/kernel/flutter_ddc_sdk_sound.dill --sound-null-safety
[  +17 ms] <- compile org-dartlang-app:/web_entrypoint.dart
[+5581 ms] lib/main.dart:7:8: Error: 'LateInitializationError' isn't a type.
[   +2 ms]   } on LateInitializationError {
[        ]        ^^^^^^^^^^^^^^^^^^^^^^^
```

`$ flutter analyze`

```logs
 bug % flutter analyze
Analyzing bug...                                                        

   info • Omit type annotations for local variables • lib/main.dart:4:3 • omit_local_variable_types
   info • Prefer final for variable declarations if they are not reassigned • lib/main.dart:4:5 • prefer_final_locals
   info • A potentially nullable type can't be used in an 'on' clause because it isn't valid to throw a nullable expression • lib/main.dart:7:8 •
          nullable_type_in_catch_clause
  error • The name 'LateInitializationError' isn't a type and can't be used in an on-catch clause • lib/main.dart:7:8 • non_type_in_catch_clause
   info • Only use double quotes for strings containing single quotes • lib/main.dart:8:11 • prefer_single_quotes
   info • Use => for short members whose body is a single return statement • lib/main.dart:16:38 • prefer_expression_function_bodies
   info • Prefer declaring const constructors on `@immutable` classes • lib/main.dart:37:3 • prefer_const_constructors_in_immutables
   info • Use => for short members whose body is a single return statement • lib/main.dart:69:38 • prefer_expression_function_bodies
   info • Prefer const with constant constructors • lib/main.dart:102:13 • prefer_const_constructors
   info • Prefer const with constant constructors • lib/main.dart:115:16 • prefer_const_constructors
   info • Sort pub dependencies • pubspec.yaml:30:3 • sort_pub_dependencies
   info • Avoid annotating types for function expression parameters • test/widget_test.dart:14:49 • avoid_types_on_closure_parameters

12 issues found. (ran in 4.4s)
```

`$ flutter doctor -v`

```logs
~ % flutter doctor -v                              
[✓] Flutter (Channel beta, 1.26.0-17.3.pre, on macOS 11.1 20C69 darwin-x64, locale en-US)
    • Flutter version 1.26.0-17.3.pre at /Users/rohit/Applications/flutter
    • Framework revision 4b50ca7f7f (15 hours ago), 2021-02-04 19:44:27 -0800
    • Engine revision 2c527d6c7e
    • Dart version 2.12.0 (build 2.12.0-259.8.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 29.0.2)
    • Android SDK at /Users/rohit/Library/Android/sdk
    • Platform android-30, build-tools 29.0.2
    • ANDROID_HOME = /Users/rohit/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.2, Build version 12B45b
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (1 available)
    • Chrome (web) • chrome • web-javascript • Google Chrome 88.0.4324.150
    ! Error: Rohit’s iPhone is not connected. Xcode will continue when Rohit’s iPhone is connected. (code -13)


```

</details>

**Comment by pedromassangocode** [comment]
Hi @tiholic 
From what I can see, the issue is related to Dart rather than to Flutter itself. Please open the issue in the dedicated [repository](https://github.com/dart-lang/sdk/issues).
Closing, as this isn't an issue with Flutter itself.  
If you disagree, please write in the comments,  
providing your `flutter doctor -v`, your `flutter run -v`,  your `pubspec.yaml`
a minimal reproducible code sample that does not use 3rd party plugins,  
and I will reopen it.
Thank you

**Comment by tiholic** [comment]
Hi @pedromassango !

I noticed this issue only after upgrading flutter beta from `1.25.0-8.3.pre` to `1.26.0-17.3.pre`

The logs I shared above are by following these steps:
1. `flutter create bug`
2. update code as mentioned in "Update the files as follows:" step _(as per my issue description)_
3. `flutter run -v`


_PS: I understand this is a dart issue, but could it be due to bad dart version shipped with `1.26.0-17.3.pre` version of flutter? (Creating a fresh project - `bug` and running with the changes above should have worked just fine otherwise)_
